superset-frontend/plugins/plugin-chart-cartodiagram/test/util/transformPropsUtil.test.ts

```diff
@@ -17,124 +17,147 @@
+      const countOfGeometries = Object.keys(result).length;
+      expect(countOfGeometries).toEqual(2);
     });
-  });
-});
 
-describe('getEchartConfigs', () => {
-  it('works for pie chart', () => {
-    const dummyTransformProps = 'dummyTransformProps';
-    const chartQueryBuilderMock: jest.MockedFunction<any> = jest.fn(
-      () => dummyTransformProps,
-    );
+    it('groups items with the correct geometry', () => {
+      const geometryColumn = 'geom';
+      const result = groupByLocation(nonTimeSeriesChartData, geometryColumn);
+      // @ts-ignore
+      const allGeom1 = result[geom1].every((item: any) => item.geom === geom1);
+      // @ts-ignore
+      const allGeom2 = result[geom2].every((item: any) => item.geom === geom2);
+      expect(allGeom1 && allGeom2).toBe(true);
+    });
+  });
 
-    const transformPropsRegistry = getChartTransformPropsRegistry();
-    transformPropsRegistry.registerValue('pie', chartQueryBuilderMock);
+  describe('getEchartConfigs', () => {
+    let chartTransformer: jest.MockedFunction<any>;
     const geomColumn = 'geom';
-
-    const chartTransformer = transformPropsRegistry.get('pie');
-
     const pieChartConfig = {
       params: {},
       viz_type: 'pie',
     };
-
     const pieChartProps: any = {
       queriesData: [{}],
     };
+    beforeEach(() => {
+      chartTransformer = jest.fn();
+    });
 
-    const result = getEchartConfigs(
-      pieChartData,
-      undefined,
-      pieChartConfig,
-      geomColumn,
-      pieChartProps,
-      chartTransformer,
-    );
-
-    const countLocations = 2;
-    expect(chartQueryBuilderMock.mock.calls).toHaveLength(countLocations);
-
-    const expectedOutput = {
-      type: 'FeatureCollection',
-      features: [
-        {
-          type: 'Feature',
-          geometry: {
-            type: 'Point',
-            coordinates: [1, 2],
-          },
-          properties: dummyTransformProps,
-        },
-        {
-          type: 'Feature',
-          geometry: {
-            type: 'Point',
-            coordinates: [3, 4],
-          },
-          properties: dummyTransformProps,
-        },
-      ],
-    };
-    expect(result).toEqual(expectedOutput);
-  });
-  it('works for bar chart', () => {
-    // TODO
-    expect(false).toEqual(true);
-  });
-  it('works for mixed time-series', () => {
-    // TODO
-    expect(false).toEqual(true);
+    it('calls the transformProps function for every location', () => {
+      getEchartConfigs(
+        nonTimeSeriesChartData,
+        undefined,
+        pieChartConfig,
+        geomColumn,
+        pieChartProps,
+        chartTransformer,
+      );
+
+      expect(chartTransformer).toHaveBeenCalledTimes(2);
+    });
+    it('returns a geojson', () => {
+      const result = getEchartConfigs(
+        nonTimeSeriesChartData,
+        undefined,
+        pieChartConfig,
+        geomColumn,
+        pieChartProps,
+        chartTransformer,
+      );
+
+      expect(result).toEqual(
```

Comment:
short comment what is happening here would be great

Reply:
See comment above